src/Reducer/CartReducer.js

```diff
@@ -0,0 +1,51 @@
+const SAVE_USER_ON_LOGIN = "SAVE_USER_ON_LOGIN"
+const GET_PRODUCTS_FROM_SERVER = "GET_PRODUCTS_FROM_SERVER"
+const ADD_TO_WISHLIST = "ADD_TO_WISHLIST"
+const REMOVE_FROM_WISHLIST = "REMOVE_FROM_WISHLIST"
+const ADD_TO_CART = "ADD_TO_CART";
+const REMOVE_FROM_CART = "REMOVE_FROM_CART";
+const SORT = "SORT";
+const TOGGLE_STOCK = "TOGGLE_STOCK";
+const TOGGLE_DELIVERY = "TOGGLE_DELIVERY";
+const SEARCH_FILTER = "SEARCH_FILTER";
+const DECREASE_CART_ITEM_QUANTITY = "DECREASE_CART_ITEM_QUANTITY";
+const CHANGE_CART_ITEM_QUANTITY = "CHANGE_CART_ITEM_QUANTITY";
+
+export function cartReducer(state, {type,payload}) {
+    switch (type) {
+        case SAVE_USER_ON_LOGIN:
+            return { ...state, cart: payload.cart, wishList: payload.wishlist }
+        case GET_PRODUCTS_FROM_SERVER:
+            return {...state, data : payload}
```

Comment:
It's recommended to send data in the payload as object

Reply:
Great Advice!! Implemented this :)